### PR TITLE
Add Matrix Vision mvBlueSirius

### PR DESCRIPTION
**Pull request by schacs**

### CODE REVIEW

**Comment by f00f** [comment]
Is this (more or less) ready to merge or WIP?

If ready: please try to get it closed soon.

If WIP: please close PR and re-open once it's ready.

**Comment by jangernert** [comment]
I wanted to test the implementation some time this week. Even have the camera set up at my desk. But basically landed in dll hell and wanted to talk to @schacs what libraries he used when writing the camera implementation.

**Comment by f00f** [comment]
Sounds like a plan.
If they're outdated we should try to update to a recent version.
In any case we should document them (so far we kept such info in our Wiki, maybe we should move it to the repo or Github wiki).

**Comment by jangernert** [comment]
There is now a dedicated `jan bullshit` folder containing all the dependencies on our server. With this and some minor fixes I got the camera running up to a point where an exception is thrown concerning EOL of the built in laser. In the native tool you can ignore the warning, but the data seems pretty much like garbage.

The next thing I'll do is contact Matrix Vision and ask for a new camera or to service the old cameras.

**Comment by jangernert** [comment]
@f00f now it's more or less ready to merge. Only basic testing has been done though.

File: BetaCameras/MatrixVision/MvBlueSirius.cpp [hunk 1]
```diff
@@ -71,94 +71,67 @@ namespace MetriCam2
 			int deviceCount = 0;
 			result = MV6D_DeviceListUpdate(h6D, &deviceCount);
 			CheckResult(result, ConnectionFailedException::typeid, 2);
-			const char* serial = { 0 };
-
-			char serialTemp[128] = { 0 };
-			int inUse = 1;
+			char serial[128] = { 0 };
+
```
Comment (f00f) [inline BetaCameras/MatrixVision/MvBlueSirius.cpp]: Remove additional blank line

**Comment by jangernert** [comment]
Yay, C style code with all of the usual mistakes, my favorite. I'll fix this tomorrow. Too tired right now.

**Comment by jangernert** [comment]
https://i.imgflip.com/261nvs.jpg

**Comment by f00f** [comment]
Nice clean-up!
* I would make `FreeImageBuffer` and `ResizeBuffer` (private?) methods on `ImageData`, probably also the `Copy*` methods
* Please rename the `Master` etc. to lowercase

**Comment by f00f** [comment]
* didn't test the icon (but looks different to how it's done in the TIVoxel project)
* `ParamDescs` are missing
* A description (i.e. comment) of the custom channels would be nice

**Comment by f00f** [comment]
`ParamDescs` must be private

**Comment by jangernert** [comment]
@f00f Icon is fixed and branch should be ready for merge. Just tested it with MetriCalibrate Intrinsics.

**Comment by jangernert** [comment]
> Why not use modelName?

I like to keep all the metadata in one place. Otherwise vendor would be a property and modelName a variable set in the constructor. So personal preference I guess.

> Could be one line: `return (nullptr == str || 0 == strcmp("", str));`

Some day I will remember to do it like this, someday. Will fix

**Pull request merged**